internal/wire/parse.go

```diff
@@ -813,13 +813,16 @@ func processStructProvider(fset *token.FileSet, info *types.Info, call *ast.Call
 		Out:      []types.Type{structPtr.Elem(), structPtr},
 	}
 	if allFields(call) {
-		provider.Args = make([]ProviderInput, st.NumFields())
+		provider.Args = make([]ProviderInput, 0)
```

Comment:
IIUC this line can be removed and it saves an allocation.

Reply:
Done.